pinot-core/src/test/java/com/linkedin/pinot/core/segment/index/loader/LoadersTest.java

```diff
@@ -147,5 +152,30 @@ public void testLoadWithStaleConversionDir()
     }
   }
 
+  @Test
+  public void testPadding()
+      throws Exception {
+    TarGzCompressionUtils.unTar(new File(TestUtils.getFileFromResourceUrl(
+        Loaders.class.getClassLoader()
+            .getResource(PADDING_OLD))), INDEX_DIR);
+    File segmentDirectory = new File(INDEX_DIR, "paddingOld");
+    SegmentMetadataImpl originalMetadata = new SegmentMetadataImpl(segmentDirectory);
+    Assert.assertEquals(originalMetadata.getColumnMetadataFor("name").getPaddingCharacter(),
+        V1Constants.Str.LEGACY_STRING_PAD_CHAR);
```

Comment:
I think you should also add tests for getting a value from the dictionary from the old segment metadata